src/main/java/seedu/address/logic/commands/consultation/ImportConsultCommand.java

```diff
@@ -0,0 +1,239 @@
+ */
+public class ImportConsultCommand extends Command {
+
+    public static final String COMMAND_WORD = "importconsult";
+    public static final CommandType COMMAND_TYPE = CommandType.IMPORTCONSULT;
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Imports consultations from a CSV file.\n"
+            + "For files in parent directory: " + COMMAND_WORD + " filename.csv\n"
+            + "For files in home directory: " + COMMAND_WORD + " ~/path/to/file.csv\n"
+            + "Examples:\n"
+            + "  " + COMMAND_WORD + " consultations.csv\n"
+            + "  " + COMMAND_WORD + " ~/documents/consultations.csv\n"
+            + "  " + COMMAND_WORD + " ~/semester1/consultations.csv";
+
+    public static final String MESSAGE_FILE_OUTSIDE_PROJECT =
+            "The file must be in the parent directory or specified with a complete path from home directory (~)";
+
+    public static final String MESSAGE_SUCCESS = "Imported %1$d consultations successfully. %2$d entries had errors.";
+    public static final String MESSAGE_SUCCESS_WITH_ERRORS = MESSAGE_SUCCESS + " Failed entries written to: %3$s";
+    public static final String MESSAGE_EMPTY_FILE = "The specified file is empty or contains no valid entries";
+    public static final String MESSAGE_INVALID_FILE = "Could not read the specified file: %1$s";
+    public static final String MESSAGE_INVALID_HEADER = "Invalid CSV header. Expected: Date,Time,Students";
+    public static final String MESSAGE_STUDENT_NOT_FOUND = "Student '%s' not found in the system";
+
+    private static final Logger logger = LogsCenter.getLogger(ImportConsultCommand.class);
+    private final Path filePath;
+    private int successCount = 0;
+    private int errorCount = 0;
+
+    /**
+     * Creates an ImportConsultCommand to import consultation data from the specified path
+     */
+    public ImportConsultCommand(String filepath) {
+        requireAllNonNull(filepath);
+        this.filePath = resolveFilePath(filepath);
+    }
+
+    @Override
+    public CommandType getCommandType() {
+        return COMMAND_TYPE;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireAllNonNull(model);
+        List<String[]> errorEntries = new ArrayList<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath.toFile()))) {
+            String header = reader.readLine();
+            if (header == null) {
+                throw new CommandException(MESSAGE_EMPTY_FILE);
+            }
+
+            if (!header.equalsIgnoreCase("Date,Time,Students")) {
+                throw new CommandException(MESSAGE_INVALID_HEADER);
+            }
+
+            String line;
+            boolean hasValidEntries = false;
+            while ((line = reader.readLine()) != null) {
+                if (line.trim().isEmpty()) {
+                    continue;
+                }
+                hasValidEntries = true;
+                try {
```

Comment:
Is it possible to extract this part into its own private method? The execute method seems a bit long.